tests/js/client/shell/api/arangosearch-memory-metrics.inc

```diff
@@ -0,0 +1,168 @@
+        if (databaseCreationOptions) {
+          db._createDatabase(dbName, databaseCreationOptions);
+        }
+        else {
+          db._createDatabase(dbName);
+        }
+        db._useDatabase(dbName);
+
+        db._create(collectionName, collectionCreationOptions);
+        generateAndInsert(collectionName);
+        db._createView(viewName, "arangosearch", {
+        links: {[collectionName]: {
+            includeAllFields: true,
+            primarySort: {"fields": [{ "field": "stringValue", "desc": true }]},
+            storedValues: ["numericValue"],
+            consolidationIntervalMsec: 100,
+            commitIntervalMsec: 100,
+          }}
+        });
+        const writers = getCompleteMetricsValues("arangodb_search_writers_memory_usage");
+        assertTrue(writers > 0);
+        const descriptors = getCompleteMetricsValues("arangodb_search_file_descriptors");
+        assertTrue(descriptors > 0, descriptors);
+      },
+      
+      tearDownAll: function () {
+        db._useDatabase('_system');
+        db._dropDatabase(dbName);
+      },
+
+      testArangoSearchMetrics: function () {
+
+        let writers_prev = getCompleteMetricsValues("arangodb_search_writers_memory_usage");
+        let writers_curr = writers_prev;
+        for (let i = 0; i < 5; i++) {
+          generateAndInsert(collectionName);
+          db._query(`for d in ${viewName} OPTIONS {waitForSync: true} LIMIT 1 RETURN 1 `);
+          writers_curr = getCompleteMetricsValues("arangodb_search_writers_memory_usage");
+          assertTrue(writers_prev < writers_curr);
+          writers_prev = writers_curr;
+        }
+
+        let readers, consolidations, descriptors, mapped, scheduler_memory, num_worker_threads;
+        for (let i = 0; i < 45; ++i) {
+
+          generateAndInsert(collectionName);
+          db._query(`for d in ${viewName} SEARCH d.numericValue == 100 RETURN d`);
+
+          [readers, consolidations, descriptors, mapped, scheduler_memory, num_worker_threads] = getCompleteMetricsValues([
+            "arangodb_search_readers_memory_usage", 
+            "arangodb_search_consolidations_memory", 
+            "arangodb_search_file_descriptors",
+            "arangodb_search_mapped_memory",
+            "arangodb_scheduler_stack_memory_usage",
+            "arangodb_scheduler_num_worker_threads"
+          ]);
+
+          assertEqual(scheduler_memory, num_worker_threads * 4 * 1000000, {scheduler_memory, num_worker_threads});
+
+          if (readers > 0 && consolidations > 0 && descriptors > 0 && (isWindows || mapped > 0)) {
+            break;
+          }
+          require("internal").sleep(0.25);
+        }
+
+        assertTrue(readers > 0);
+        assertTrue(consolidations > 0);
+        assertTrue(descriptors > 0);
+        if (!isWindows) {
+          assertTrue(mapped > 0);
+        }
+
+        {
+          const oldValue = getCompleteMetricsValues("arangodb_search_writers_memory_usage");
```

Comment:
This could be flaky if commit will arrive and actually process removes before line 138 - memory would be released. Suggest to set commitIntervalMsec to 0 for the view before line 137. And reset it back after line 139

Reply:
Done